mantaray/backend.py

```diff
@@ -200,7 +205,16 @@ def _connect_loop(self) -> None:
                 continue
 
             if self.password is not None:
-                self.send("CAP REQ sasl")
+                self.cap_req.append("sasl")
```

Comment:
Presumably `cap_req` and `cap_list` should be cleared somewhere in the middle of `_connect_loop`, before you append more capabilities. It is currently done when receiving `RPL_WELCOME`, but there's no way that code in different files will stay in sync when we add a new list that also needs to be cleared.